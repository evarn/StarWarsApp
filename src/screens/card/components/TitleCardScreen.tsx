import React from 'react';
import {StyleSheet, Text} from 'react-native';
import Strings from '../../../constants/strings';
import Colors from '../../../constants/colors';
import Fonts from '../../../constants/fonts';

interface ITitleCardScreen {
  count: number;
}

const TitleCardScreen = ({count}: ITitleCardScreen) => {
  return (
    <Text style={styles.viewTitle}>
      <Text style={[styles.textTitle, styles.textTitleBold]}>{count}</Text>
      <Text style={[styles.textTitle, styles.textTitleBold]}>
        {Strings.CHARACTERS_TITLE_TEXT_1}
      </Text>
      <Text style={styles.textTitle}>{Strings.CHARACTERS_TITLE_TEXT_2}</Text>
    </Text>
  );
};

const styles = StyleSheet.create({
  viewTitle: {
    marginTop: 8,
    textAlign: 'center',
  },
  textTitle: {
    color: Colors.BLACK_1,
    fontFamily: Fonts.PRIMARY,
    fontStyle: 'normal',
    fontWeight: '400',
    fontSize: 30,
    letterSpacing: 3,
  },
  textTitleBold: {
    fontWeight: '700',
  },
});

export default TitleCardScreen;
