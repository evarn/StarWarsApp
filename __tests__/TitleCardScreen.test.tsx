import 'react-native';
import React from 'react';
import {render, cleanup} from '@testing-library/react-native';
import TitleCardScreen from '../src/screens/card/components/TitleCardScreen';

afterEach(cleanup);

describe('TitleCardScreen', () => {
  it('TitleCardScreen snapshoot', () => {
    const {toJSON} = render(<TitleCardScreen count={0} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('TitleCardScreen text', () => {
    const {getByText} = render(<TitleCardScreen count={0} />);
    const text = 'for you to choose your favorite';
    const foundText = getByText(text);
    expect(foundText.props.children).toEqual(text);
  });
});
