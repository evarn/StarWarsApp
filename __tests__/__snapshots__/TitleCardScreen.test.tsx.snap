// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleCardScreen TitleCardScreen snapshoot 1`] = `
<Text
  style={
    {
      "marginTop": 8,
      "textAlign": "center",
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#000000",
          "fontFamily": "Karla",
          "fontSize": 30,
          "fontStyle": "normal",
          "fontWeight": "400",
          "letterSpacing": 3,
        },
        {
          "fontWeight": "700",
        },
      ]
    }
    testID="TitleCardScreenText"
  >
    0
  </Text>
  <Text
    style={
      [
        {
          "color": "#000000",
          "fontFamily": "Karla",
          "fontSize": 30,
          "fontStyle": "normal",
          "fontWeight": "400",
          "letterSpacing": 3,
        },
        {
          "fontWeight": "700",
        },
      ]
    }
  >
     Peoples 
  </Text>
  <Text
    style={
      {
        "color": "#000000",
        "fontFamily": "Karla",
        "fontSize": 30,
        "fontStyle": "normal",
        "fontWeight": "400",
        "letterSpacing": 3,
      }
    }
  >
    for you to choose your favorite
  </Text>
</Text>
`;
